src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+  }
 }
 
+Animal.alive = []; // Wouldn't allow me to put it inside >~<
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (!(animal instanceof Herbivore) || animal.hidden === true) {
+      return;
+    }
+
+    animal.health -= 50;
```

Comment:
What's the purpose of empty return? Better just remove the health only for true condition